### ISSUE TITLE
knex.dropTable() errors on Oracle when .withSchema()

### ISSUE THREAD

**Issue by code-ape** [knex/knex#4595]
# Environment

**Knex version:** `0.95.8`
**Database + version:** Oracle 18c (@atiertant)
**OS:** Ubuntu Linux

# Bug

1. **Explain what kind of behaviour you are getting and how you think it should do**

When attempting to delete a table for a given schema (called a "user" in Oracle) then the migration fails with an error.

2. Error message

```sql
-- indent formatted by issue filer for easier reading
drop table "my_table";
begin
  execute immediate 'drop sequence "my_table"';
  exception when others then if sqlcode != -2289 then raise;
  end if;
end;

-- Error: drop table "my_table" - ORA-00942: table or view does not exist
```

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

```ts
// Create schema / user
const sanitizedRaw = knex.raw(`CREATE USER ?? IDENTIFIED BY ??`, [schemaName, schemaName]).toQuery()
await knex.schema.raw(sanitizedRaw)
// Create table
await knex.schema.withSchema('my_schema').createTable('my_table', (table) => {
  table.increments('id').primary()
})
// Drop table
knex.schema.withSchema('my_schema').dropTable('my_table')
```

**Comment by code-ape** [comment]
It appears this can be resolved by fixing the following functions in [`knex/lib/dialects/oracle/schema/oracle-compiler.js`](https://github.com/knex/knex/blob/124107e0eca5073049b10fb3d8628723e5505cd5/lib/dialects/oracle/schema/oracle-compiler.js):

1. `dropTable` needs to use schema name if it exists (`this.schema`).
1. `dropTableIfExists` needs to use schema name if it exists (`this.schema`).
1. `dropSequenceIfExists` needs to use schema name if it exists (`this.schema`).

Example code:

```ts
  dropSequenceIfExists(sequenceName) {
    const prefix = this.schema ? `"${this.schema}".` : "";
    this.pushQuery(
      utils.wrapSqlWithCatch(
        `drop sequence ${prefix}${this.formatter.wrap(sequenceName)}`,
        -2289
      )
    );
  }
```

**Comment by kibertoad** [comment]
Released in 0.95.9
